pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -3636,6 +3638,13 @@ private PersistentReplicator getReplicatorReference(String replName, PersistentT
                         p -> new PartitionedTopicMetadata(numPartitions));
                 updatePartition.complete(null);
             } catch (Exception e) {
+                for(int i = oldPartition; i < numPartitions; i++) {
+                    try {
+                        namespaceResources().getPartitionedTopicResources().delete(ZkAdminPaths.managedLedgerPath(topicName.getPartition(i)));
```

Comment:
It's similar to my previous comment here. Can we avoid waiting a `CompletableFuture` in future's callback?

Reply:
`delete` is a asynchronous method